### PR TITLE
now account IDs get saved to and loaded from localStorage

### PR DESCRIPTION
**Pull request by ryancwalsh**
https://near-foundation.atlassian.net/jira/software/c/projects/TTA/boards/5?modal=detail&selectedIssue=TTA-46

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -1,18 +1,46 @@
...
+// eslint-disable-next-line max-lines-per-function
 export default function Home() {
+  // `accountIds` is an array instead of a Set so that users can temporarily paste multiple lines of the same account ID and then edit them to be unique before submitting the form. If they were a Set, pasting an account ID more than once would result in confusing behavior of the form just showing the first paste.
+  const [accountIds, setAccountIds] = useState<AccountId[]>(initialAccountIds);
+  // See comment in `onSubmit` about why not to use just `useLocalStorage`.
+  const [accountIdsLocalStorage, setAccountIdsLocalStorage] = useLocalStorage<AccountId[]>('accountIds', initialAccountIds); // useLocalStorage seemed unable to save Set<AccountId>.
+
+  function onChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+    const accountIdsArray = event.target.value.split('\n').map((accountId) => accountId.trim());
+    console.log('onChange', { accountIdsArray });
```
Comment (PierreLeGuen) [inline pages/index.tsx:30]: Remove console logs

File: pages/index.tsx [hunk 2]
```diff
@@ -1,18 +1,46 @@
...
 defaultStartDate.setDate(defaultStartDate.getDate() - 30); // 30 days ago
 const defaultStartDateString = defaultStartDate.toISOString().slice(0, 10);
 const defaultEndDate = new Date();
 defaultEndDate.setDate(defaultEndDate.getDate() + 1); // tomorrow
 const defaultEndDateString = defaultEndDate.toISOString().slice(0, 10);
+const initialAccountIds: AccountId[] = ACCOUNT_IDS ? ACCOUNT_IDS.split(',') : [];
 
+// eslint-disable-next-line max-lines-per-function
 export default function Home() {
+  // `accountIds` is an array instead of a Set so that users can temporarily paste multiple lines of the same account ID and then edit them to be unique before submitting the form. If they were a Set, pasting an account ID more than once would result in confusing behavior of the form just showing the first paste.
```
Comment (PierreLeGuen) [inline pages/index.tsx]: Could we spread long comments on multiple lines?

File: helpers/localStorage.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+  });
+  // Return a wrapped version of useState's setter function that ...
+  // ... persists the new value to localStorage.
+  const setValue = (value: T | ((value_: T) => T)) => {
+    try {
+      // Allow value to be a function so we have same API as useState
+      const valueToStore = value instanceof Function ? value(storedValue) : value;
+      // Save state
+      console.log(`Saving to localStorage key ${key}... 'valueToStore' = ${valueToStore}`);
```
Comment (PierreLeGuen) [inline helpers/localStorage.ts:32]: Remove console log

**Review by PierreLeGuen — CHANGES_REQUESTED** [review]

File: pages/index.tsx (hunk 2 above)
Comment (ryancwalsh) [reply pages/index.tsx]: Yes. Done, and added an eslint rule to enforce it going forward.

**Pull request merged**